pkg/scalers/azure/azure_aad_workload_identity.go

```diff
@@ -124,6 +126,48 @@ func (aadWiConfig ADWorkloadIdentityConfig) Authorizer() (autorest.Authorizer, e
 		aadWiConfig.ctx, aadWiConfig.IdentityID, aadWiConfig.Resource)), nil
 }
 
+// ADWorkloadIdentityCredential is a type that implements the TokenCredential interface.
+// Once azure-sdk-for-go supports Workload Identity we can remove this and use default implementation
+// https://github.com/Azure/azure-sdk-for-go/issues/15615
+type ADWorkloadIdentityCredential struct {
+	ctx        context.Context
+	IdentityID string
+	Resource   string
+	aadToken   AADToken
+}
```

Comment:
Unfortunate that we cannot reuse the `ADWorkloadIdentityTokenProvider` type even though they have the exact same properties, because they need to implement different interfaces which have the `GetToken` method.

Reply:
yeah, but I hope we can remove this new implementation soon with the official support from the SDK